### PR TITLE
Intro paras for Freezing and Packaging your code

### PR DESCRIPTION
**Pull request by tartley**
I've tried to describe why you would choose freezing versus packaging versus creating a Linux distribution package, and the relative advantages and disadvantages of each method.

My intent is to add more content on the Freezing page once this commit is accepted, so I can hear your feedback before doing any further work.

### CODE REVIEW

File: docs/shipping/freezing.rst [hunk 1]
```diff
@@ -1,17 +1,36 @@
...
-- Dropbox
-- BitTorrent
-- ...
+The advantage of distributing this way is that your application will work even
+if the user doesn't already have the required version of Python installed. On
+Windows, and even on many Linux distributions and OSX versions, the right
+version of Python will not already be installed.
 
-.. todo:: Fill in "Freezing Your Code" stub
+One disadvantage is that it will bloat your distribution by about 2Mb.
```
Comment (adrianmoisey) [inline docs/shipping/freezing.rst]: `MB` or `Mb`?

Comment (tartley) [inline docs/shipping/freezing.rst]: Personally I use 'Mb', but I'm very happy to use MB if that's preferred.

Comment (tartley) [inline docs/shipping/freezing.rst]: Yep, fair enough, Wikipedia agrees with you. Commit incoming...

**Comment by tartley** [comment]
I guess with hindsight, I've not remotely made it clear that packaging is pretty much a prerequisite for both freezing and creating a .deb/rpm. I'm also unhappy about not making explaining that you might choose to package, and then optionally freeze, and then create a Linux distribution package from that.

Personally I think we should merge this PR, then address those issues separately. But I could understand if you feel we should fix the overall structural issues before we merging in great wads of text. Let me know and I'd be happy to close this PR without merging, and try again.

**Comment by tartley** [comment]
Is there a less ambiguous term I should use for .deb/rpm files other than "Linux distribution package"? I don't like the overloaded meanings of "package".

File: docs/shipping/freezing.rst [hunk 2]
```diff
@@ -1,17 +1,36 @@
...
+
+Alternatives to Freezing
+------------------------
 
+:ref:`Packaging your code <packaging-your-code-ref>` is for distributing
+libraries or tools to other developers.
+
+On Linux, an alternative to freezing is to
+:ref:`create a Linux distro package <packaging-for-linux-distributions-ref>`
+(e.g. a .deb file for Debian or Ubuntu.)
```
Comment (htgoebel) [inline docs/shipping/freezing.rst]: Please mention .rpm here, too.

File: docs/shipping/packaging.rst [hunk 3]
```diff
@@ -106,9 +136,42 @@ prerequisite for this is that you have an Amazon AWS account with an S3 bucket.
...
+
+Creating a distribution package is also problematic if your code is for a
+version of Python that isn't currently supported by a distribution.
+Having to tell *some versions* of Ubuntu end-users that they need to add `the
+'dead-snakes' PPA <https://launchpad.net/~fkrull/+archive/ubuntu/deadsnakes>`_
+using `sudo apt-repository` commands before they can install your .deb file
+makes for an extremely hostile user experience. Not only that, but you'd have
+to maintain a custom equivalent of these instructions for every distribution,
+and worse, have your users read, understand, and act on them.
+
```
Comment (htgoebel) [inline docs/shipping/packaging.rst]: You may want to describe `bdist_rpm` here, too.

**Comment by kennethreitz** [comment]
many thanks!

**Pull request merged**